kyaml/runfn/runfn.go

```diff
@@ -468,11 +468,26 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 		if err != nil {
 			return nil, err
 		}
+
+		// Storage mounts can either come from kustomize fn run --mounts,
+		// or from the declarative function mounts field.
+		// In the latter case, we ensure that the mount path is subject
+		// to kustomize's load restrictions.
+		storageMounts := spec.Container.StorageMounts
+		for _, mount := range storageMounts {
+			if strings.HasPrefix(filepath.Clean(mount.Src), "../") {
+				return nil, fmt.Errorf("mount path %s is not permitted; declarative mount paths must be under the current directory", mount.Src) //nolint:goerr113
+			}
+		}
+		if len(storageMounts) == 0 {
```

Comment:
So declarative always overrides imperative? I think I'd expect it to be the other way around, or for imperative to be appended. I'm guessing we don't document one way or the other, currently.

Reply:
I'll change the logic to append it. That sounds like more intuitive behavior to me.